libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1007,17 +1064,20 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      *   updates is received. See [RoadGraphDataUpdateCallback].
      */
     fun requestRoadGraphDataUpdate(callback: RoadGraphDataUpdateCallback) {
-        CacheHandleWrapper.requestRoadGraphDataUpdate(navigator.cache, callback)
+        runOnNavigationLooper {
+            CacheHandleWrapper.requestRoadGraphDataUpdate(navigator.cache, callback)
+        }
     }
 
     private fun internalSetNavigationRoutes(
         routes: List<NavigationRoute>,
         setRoutesInfo: SetRoutes,
         callback: RoutesSetCallback? = null,
     ) {
+        threadController.assertSDKThread()
```

Comment:
Is it required? It seems like if it's not, it's an internal error. I'd suggest not crashing in this case but using another way to find this early (in a test, for example).